### PR TITLE
Expose rendering teardown event

### PR DESCRIPTION
**Pull request by hidmic**
# 🎉 New feature

Follow-up after #1475. Turns out rendering sensor destruction must also occur in the rendering thread.

## Summary

This patch adds a `rendering::RenderTeardown` event for downstream users to perform cleanup upon the rendering thread stopping. 

## Test it

...
## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by arjo129 — APPROVED** [review]
LGTM

**Comment by hidmic** [comment]
@arjo129 small change to ensure custom rendering sensors that depend on builtin rendering sensors get destroyed first.

**Review by iche033 — APPROVED** [review]
just one minor comment otherwise looks good to me.

File: include/gz/sim/rendering/Events.hh [hunk 1]
```diff
@@ -77,6 +77,18 @@ namespace gz
       using PostRender = gz::common::EventT<void(void),
           struct PostRenderTag>;
 
+      /// \brief The render teardown event is emitted right before the
+      /// rendering thread is torn down. The veven is emitted in the
```
[the remark this answers is not in the document]
Comment (hidmic) [reply include/gz/sim/rendering/Events.hh]: Fixed in https://github.com/gazebosim/gz-sim/pull/1539/commits/2a477f5e1fb502c70f161a01b10e621086ea8cad.

**Comment by hidmic** [comment]
@iche033 @arjo129 CI failures have nothing to do with this patch. How shall we go about this?

**Comment by chapulina** [comment]
> CI failures have nothing to do with this patch. How shall we go about this?

Agreed. Feel free to merge, CI is unstable for this branch right now, no need to block this PR.

**Pull request merged**